src/js/components/TaskView.js

```diff
@@ -133,55 +137,74 @@ class TaskView extends mixin(SaveStateMixin) {
   }
 
   getTaskTable(tasks) {
+    let {parentRouter, inverseStyle} = this.props;
     return (
-      <TaskTable tasks={tasks} parentRouter={this.props.parentRouter} />
+      <TaskTable tasks={tasks} parentRouter={parentRouter}
+        inverseStyle={inverseStyle} />
     );
   }
 
-  getHeaderText(tasks) {
-    let currentStatus = this.state.filterByStatus;
-    let tasksLength = tasks.length;
-    if (currentStatus === 'all') {
-      return `${tasksLength} ${StringUtil.pluralize('Task', tasksLength)}`;
-    }
-
-    const displayNameMap = {
-      active: 'Active',
-      completed: 'Completed'
-    };
+  resetFilter() {
+    this.setState({
+      searchString: '',
+      filterByStatus: 'all'
+    });
+  }
 
-    let statusCount = this.getStatusCounts(tasks)[currentStatus];
-    let displayName = displayNameMap[currentStatus];
-    let pluralizedTasks = StringUtil.pluralize('Task', statusCount);
-    return `${statusCount} ${displayName} ${pluralizedTasks}`;
+  getButtonContent(filterName, count) {
+    return (
+      <span className="button-align-content">
+        <span className="label">{StringUtil.capitalize(filterName)}</span>
+        <span className="badge">{count || 0}</span>
+      </span>
+    );
   }
 
   getContent() {
-    let state = this.state;
-    let tasks = this.props.tasks || [];
+    let {searchString, filterByStatus} = this.state;
+    let {tasks, inverseStyle} = this.props;
+    let totalNumberOfTasks = tasks.length;
 
-    if (state.searchString !== '') {
-      tasks = StringUtil.filterByString(tasks, 'name', state.searchString);
+    // Get task states based on TaskStates types
+    let taskStates = tasks.map(function (task) {
+      let {stateTypes} = TaskStates[task.state];
+
+      return stateTypes.find(function (state) {
+        return state === 'active' || state === 'completed';
+      });
+    });
+
+    if (searchString !== '') {
+      tasks = StringUtil.filterByString(tasks, 'name', searchString);
     }
 
     tasks = this.filterByCurrentStatus(tasks);
 
     return (
       <div className="flex-container-col flex-grow">
-        <span className="h4 text-align-left flush-top">
-          {this.getHeaderText(tasks)}
-        </span>
-        <div className="flex-box control-group">
-          <FilterInputText
-            searchString={state.searchString}
-            handleFilterChange={this.handleSearchStringChange}
-            inverseStyle={false} />
-          <div>
-            <FilterByTaskState
-              statuses={this.getStatuses(tasks)}
-              handleFilterChange={this.handleStatusFilterChange}
-              totalTasksCount={tasks.length}
-              currentStatus={state.filterByStatus}/>
+        <FilterHeadline
+          inverseStyle={inverseStyle}
+          onReset={this.resetFilter}
+          name="Tasks"
+          currentLength={tasks.length}
+          totalLength={totalNumberOfTasks} />
+        <div className="filter-bar">
```

Comment:
But it will only replace the div, right? Anyway, let's refactor this in a different PR.